### PR TITLE
feat(sink): async truncate for kinesis, pulsar, nats, redis and clickhouse sink

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Extract the common logic of async truncate of kafka sink out to a common `AsyncTruncateSinkWriter`, which has a `write_chunk` method that handles a stream and may add a delivery future to the future manager.

The common `AsyncTruncateSinkWriter` is ported to the kinesis, pulsar, nats and clickhouse sink.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/connector/src/sink/nats.rs [hunk 1]
```diff
@@ -110,11 +118,11 @@ impl Sink for NatsSink {
         Ok(())
     }
 
-    async fn new_log_sinker(&self, writer_param: SinkWriterParam) -> Result<Self::LogSinker> {
+    async fn new_log_sinker(&self, _writer_param: SinkWriterParam) -> Result<Self::LogSinker> {
         Ok(
             NatsSinkWriter::new(self.config.clone(), self.schema.clone())
                 .await?
-                .into_log_sinker(writer_param.sink_metrics),
+                .into_log_sinker(usize::MAX),
```
Comment (yufansong) [inline src/connector/src/sink/nats.rs:125]: Why we change this into `usize:MAX` ?

**Review by yufansong — APPROVED** [review]
Nats part LGTM

File: src/connector/src/sink/nats.rs (hunk 1 above)
Comment (wenym1) [reply src/connector/src/sink/nats.rs:125]: Just a random value. We won't add any delivery future for nats sink.

**Review by xzhseh — APPROVED** [review]
Kafka part LGTM.

**Review by Rossil2012 — APPROVED** [review]
Pulsar part LGTM.

**Review by xxhZs — APPROVED** [review]
LGTM

**Pull request merged**